src/color.rs

```diff
@@ -386,47 +387,84 @@ fn clamp_f32(val: f32) -> u8 {
 
 #[inline]
 fn parse_color_function(name: &str, arguments: &mut Parser) -> Result<Color, ()> {
-    let (is_rgb, has_alpha) = match_ignore_ascii_case! { name,
-        "rgba" => (true, true),
-        "rgb" => (true, false),
-        "hsl" => (false, false),
-        "hsla" => (false, true),
+    let is_rgb = match_ignore_ascii_case! { name,
+        "rgba" => true,
+        "rgb" => true,
+        "hsl" => false,
+        "hsla" => false,
         _ => return Err(())
     };
 
     let red: u8;
     let green: u8;
     let blue: u8;
+    let mut uses_commas = false;
     if is_rgb {
         // Either integers or percentages, but all the same type.
         // https://drafts.csswg.org/css-color/#rgb-functions
         match try!(arguments.next()) {
-            Token::Number(NumericValue { int_value: Some(v), .. }) => {
-                red = clamp_i32(v);
-                try!(arguments.expect_comma());
-                green = clamp_i32(try!(arguments.expect_integer()));
-                try!(arguments.expect_comma());
-                blue = clamp_i32(try!(arguments.expect_integer()));
+            Token::Number(NumericValue { value: v, .. }) => {
+                red = clamp_i32(v as i32);
+                green = clamp_i32(match try!(arguments.next()) {
+                    Token::Number(NumericValue { value: v, .. }) => v,
+                    Token::Comma => {
+                        uses_commas = true;
+                        try!(arguments.expect_number())
+                    }
+                    _ => return Err(())
+                } as i32);
+                if uses_commas {
+                    try!(arguments.expect_comma());
+                }
+                blue = clamp_i32(try!(arguments.expect_number()) as i32);
             }
             Token::Percentage(ref v) => {
                 red = clamp_f32(v.unit_value);
-                try!(arguments.expect_comma());
-                green = clamp_f32(try!(arguments.expect_percentage()));
-                try!(arguments.expect_comma());
+                green = clamp_f32(match try!(arguments.next()) {
+                    Token::Percentage(ref v) => v.unit_value,
+                    Token::Comma => {
+                        uses_commas = true;
+                        try!(arguments.expect_percentage())
+                    }
+                    _ => return Err(())
+                });
+                if uses_commas {
+                    try!(arguments.expect_comma());
+                }
                 blue = clamp_f32(try!(arguments.expect_percentage()));
             }
             _ => return Err(())
         };
     } else {
-        let hue_degrees = try!(arguments.expect_number());
+        let hue_degrees = match try!(arguments.next()) {
+                Token::Number(NumericValue { value: v, .. }) => v,
+                Token::Dimension(NumericValue { value: v, .. }, unit) => {
+                    match &*unit {
```

Comment:
This needs to use case insensitive matching, please use `match_ignore_ascii_case!`. Spec: https://drafts.csswg.org/css-values/#dimensions

> Like keywords, unit identifiers are ASCII case-insensitive